dietpi/dietpi-software

```diff
@@ -4005,36 +4039,32 @@ Package: openssl libssl*\nPin: origin packages.sury.org\nPin-Priority: -1' > /et
 			G_AGI "${apackages[@]}"
 			systemctl -q is-active "$PHP_NAME-fpm" && G_EXEC systemctl stop "$PHP_NAME-fpm"
 
-			# PHP-FPM
-			if (( ${aSOFTWARE_INSTALL_STATE[84]} > 0 || ${aSOFTWARE_INSTALL_STATE[85]} > 0 )); then
-
-				# Base
-				sed -i '/cgi.fix_pathinfo=/c\cgi.fix_pathinfo=1' "$FP_PHP_BASE_DIR/fpm/php.ini"
+			# Assure that mod_php is purged in favour of PHP-FPM
+			G_AGP 'libapache2-mod-php*'
+			G_EXEC rm -Rf /{etc/php,var/lib/php/modules}/*/apache2
 
+			# PHP-FPM
+			if (( ${aSOFTWARE_INSTALL_STATE[83]} > 0 || ${aSOFTWARE_INSTALL_STATE[84]} > 0 || ${aSOFTWARE_INSTALL_STATE[85]} > 0 ))
+			then
 				# Optimisations based on total cores
-				sed -i "/pm.max_children = /c\pm.max_children = $(( $G_HW_CPU_CORES * 3 ))" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
-				sed -i "/pm.start_servers = /c\pm.start_servers = $G_HW_CPU_CORES" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
-				sed -i "/pm.min_spare_servers = /c\pm.min_spare_servers = $G_HW_CPU_CORES" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
-				sed -i "/pm.max_spare_servers = /c\pm.max_spare_servers = $G_HW_CPU_CORES" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
-
-				# Environment settings
-				sed -i "/env\[HOSTNAME\]/c\env\[HOSTNAME\] = \$HOSTNAME" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
-				sed -i "/env\[PATH\]/c\env\[PATH\] = /usr/local/bin:/usr/bin:/bin" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
-
+				G_CONFIG_INJECT 'pm.max_children[[:blank:]=]' "pm.max_children = $(( $G_HW_CPU_CORES * 3 ))" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
+				G_CONFIG_INJECT 'pm.start_servers[[:blank:]=]' "pm.start_servers = $G_HW_CPU_CORES" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
+				G_CONFIG_INJECT 'pm.min_spare_servers[[:blank:]=]' "pm.min_spare_servers = $G_HW_CPU_CORES" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
+				G_CONFIG_INJECT 'pm.max_spare_servers[[:blank:]=]' "pm.max_spare_servers = $G_HW_CPU_CORES" "$FP_PHP_BASE_DIR/fpm/pool.d/www.conf"
```

Comment:
Generally I'd prefer using a static amount of FPM workers here as well to avoid constant spawning/destroying of processes. I have 12 workers running on `dietpi.com` and while sometimes during peaks all 12 are used, webserver requests are then simply queried up to a defined maximum query length, `511` by default (much higher on `dietpi.com`). And since with OPcache, PHP is usually fast, even queried requests are usually handled within fractions of a second.

But OPcache is shared across all PHP-FPM workers, so for keeping it efficiently used, static workers are not required, and I'm not sure whether spawning and destroying of additional workers is something that significantly impacts performance or CPU load in common scenarios. Also of course we do not know the expected traffic of the installed instance, hence having a range of FPM workers covers a larger range of use cases OOTB without having either too many or too few processes up most of the time. So I'll leave it untouched for now.